lib/core-classes/AssetCategoryService.ts

```diff
@@ -20,5 +34,38 @@ export class AssetCategoryService {
     }
     return metadataList;
   }
+
+  getMetadatasValues (assetCategory : AssetCategoryContent) {
+    let metadataValueList;
+    if (! assetCategory.assetMetadatas) {
+      metadataValueList = [];
+    }
+    else {
+      metadataValueList = JSON.parse(JSON.stringify(assetCategory.metadatasValues));
+    }
+    let assetCategoryTmp = assetCategory;
+    while (assetCategoryTmp.parent) {
+      if (assetCategoryTmp.parent.metadatasValues) {
+        metadataValueList = metadataValueList.concat(assetCategoryTmp.parent.metadatasValues);
+      }
+      assetCategoryTmp = assetCategoryTmp.parent;
+    }
+    return metadataValueList;
+  }
+
+
+  async validateMetadata (assetMetadata, engineId, category) {
+    const assetCategory = await this.sdk.document.get<AssetCategoryContent>(engineId, 'asset-category', category);
+    const metadatas = this.getMetadatas(assetCategory._source);
+    const metadatasValues = this.getMetadatasValues(assetCategory._source);
+    for (const metadata of metadatas) {
+      if (metadata.mandatory) {
+        // eslint-disable-next-line no-prototype-builtins
+        if (! (assetMetadata && assetMetadata.hasOwnProperty(metadata.name)) && ! metadatasValues.hasOwnProperty(metadata.name)) {
```

Comment:
Why are you not using the bracket notation?
Also, `assetMetadata` should not null here, as far as I understand, this method should not be called with a null value for `assetMetadata` so an error should be raised earlier
```suggestion
        if (! (assetMetadata && assetMetadata[metadata.name]) && ! metadatasValues[metadata.name]) {
```

Reply:
done